lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformNumberFormatterICU.java

```diff
@@ -0,0 +1,358 @@
+package com.facebook.hermes.intl;
+
+import android.icu.text.CompactDecimalFormat;
+import android.icu.text.DecimalFormat;
+import android.icu.text.DecimalFormatSymbols;
+import android.icu.text.MeasureFormat;
+import android.icu.text.NumberingSystem;
+import android.icu.util.Currency;
+import android.icu.util.Measure;
+import android.icu.util.MeasureUnit;
+import android.icu.util.ULocale;
+
+import java.math.BigDecimal;
+import java.text.AttributedCharacterIterator;
+import java.util.ArrayList;
+
+import static android.icu.text.NumberFormat.SCIENTIFICSTYLE;
+import static com.facebook.hermes.intl.IPlatformNumberFormatter.Style.CURRENCY;
+
+// An implementation of NumberFormat services using ICU4J APIs available in Android from API 24 onwards.
+// We could have a much richer implementation on newer API versions with // Note :: https://developer.android.com/reference/android/icu/number/NumberFormatter which was introduced in API 30.
+//
+// The implementation has at least the following deficiencies,
+// 1. SignDisplay attribute implementation is partly hacky, and doesn't work in many cases.
+// 2. formatToParts does report the whole formatted string as a "literal"
+public class PlatformNumberFormatterICU implements IPlatformNumberFormatter {
+
+    private java.text.Format mFinalFormat;
+    private DecimalFormat mDecimalFormat;
+    private ILocaleObject mLocaleObject;
+    private IPlatformNumberFormatter.Style mStyle;
+
+    private MeasureUnit mMeasureUnit = null;
+    private String mUnitId = null;
+
+    private PlatformNumberFormatterICU(DecimalFormat decimalFormat, ILocaleObject localeObject, IPlatformNumberFormatter.Style style) {
+        mDecimalFormat = decimalFormat;
+        mFinalFormat = decimalFormat;
+        mLocaleObject = localeObject;
+        mStyle = style;
+
+        mDecimalFormat.setRoundingMode(BigDecimal.ROUND_HALF_UP);
+    }
+
+    @Override
+    public PlatformNumberFormatterICU configureCurrency(String currencyCode, IPlatformNumberFormatter.CurrencyDisplay currencyDisplay) throws JSRangeErrorException {
+        if (mStyle == CURRENCY) {
+
+            Currency currency = Currency.getInstance(currencyCode);
+            mDecimalFormat.setCurrency(currency);
+
+            String currencySymbol;
+            switch (currencyDisplay) {
+                case NAME:
+                    currencySymbol = currency.getName((ULocale) mLocaleObject.getLocale(), Currency.LONG_NAME, null);
+                    break;
+                case CODE:
+                    currencySymbol = currencyCode;
+                    break;
+                case SYMBOL:
+                case NARROWSYMBOL:
+                default:
+                    currencySymbol = currency.getName((ULocale) mLocaleObject.getLocale(), Currency.SYMBOL_NAME, null);
+                    break;
+            }
+
+            mDecimalFormat.getDecimalFormatSymbols().setCurrencySymbol(currencySymbol);
```

Comment:
getDecimalFormatSymbols is documented to return a copy of the symbols, so this seems like it shouldn't work.  Has it been tested?

Reply:
My assumption was that it returns a reference.. I guess it was not tested .. I've changed the code to set the updated symbols back to the format object ..